java/ql/src/Security/CWE/CWE-200/TempDirLocalInformationDisclosure.ql

```diff
@@ -193,6 +192,8 @@ abstract class MethodAccessInsecureFileCreation extends MethodAccess {
    * Gets the type of entity created (e.g. `file`, `directory`, ...).
    */
   abstract string getFileSystemEntityType();
+
+  DataFlow::Node getNode() { result.asExpr() = this }
```

Comment:
Is this needed because `MethodAccessInsecureFileCreation` is used as the type parameter of `PathGraphSig`? I suppose we can't make `MethodAccessInsecureFileCreation` implement `PathNodeSig` in some way to enforce that, can we?

Reply:
Correct.